crates/modules/logger/src/lib.rs

```diff
@@ -119,23 +148,30 @@ impl Logger {
             Err(Self {
                 console,
                 syslog: opt_sock,
+                output_format,
             })
         } else {
             Ok(Self {
                 console,
                 syslog: opt_sock,
+                output_format,
             })
         }
     }
 
     fn process(&mut self, event: &Event) -> io::Result<()> {
         if event.header().threat.is_some() {
+            let out = match self.output_format {
+                OutputFormat::Plaintext => format!("{:#}", event),
+                OutputFormat::Json => serde_json::to_string(event)?,
+            };
+
             if self.console {
-                println!("{:#}", event);
+                println!("{}", out);
             }
 
             if let Some(ref mut syslog) = &mut self.syslog {
-                syslog.send(format!("{}", event).as_bytes())?;
+                syslog.send(out.as_bytes())?;
```

Comment:
before the changes console was using the alternate form of printing `{:#}` while syslog the normal `{}`

Reply:
To make also the json lazy without computing it multiple times, you can try something like this:

```rust
// ...
            let json_event = OnceCell::new();

            let json_event = || -> Result<&str,_> { json_event.get_or_init(|| serde_json::to_string(event)?) };

            if self.console {
               let out = match self.output_format {
                OutputFormat::Plaintext => format!("{event:#}"),
                OutputFormat::Json => json_event()?,
               };
                println!("{out}");
            }
        
            if let Some(ref mut syslog) = &mut self.syslog {
               let out = match self.output_format {
                  OutputFormat::Plaintext => format!("{event}"),
                  OutputFormat::Json => json_event()?,
               };
               syslog.send(out.as_bytes())?;
            }

// ... 
```

But you have to check if it compiles, probably I made some mistakes

Reply:
Thanks! can you open an issue for the other bug?

Reply:
Perfect!!